### PR TITLE
Resizing embedding matrix before sending it to the optimizer.

### PR DESCRIPTION
**Pull request by ngarneau**
Hi there,

This is a minor bug when fine-tuning a pre-trained language model with a larger vocabulary using the run_lm_finetuning.py script.
Nothing major but I spent a couple of hours trying to figure out why my token embeddings were not being updated with their corresponding gradient :)

This bug will rise when you add new token types in the Tokenizer.
Since the resizing was done after passing the params to the optimizer, the wrong set of params for the embedding table were optimized.

Cheers

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
LGTM! Thanks for taking the time to look into this!
